### PR TITLE
Add copyright control

### PR DESCRIPTION
**Pull request by oterral**
# How to

# Others

- [x] It's not a hack or at least an unauthorized hack :).
- [x] The images added are optimized.
- [x] Everything in ticket description has been fixed.
- [x] The author of the MR has made its own review before assigning the reviewer.
- [x] The title means something for a human being.
- [x] The title contains [WIP] if it's necessary.
- [x] Labels applied. if it's a release? a hotfix?
- [x] The new class' members & methods are well documented
- [x] Tests added.

### CODE REVIEW

File: src/common/utils/removeDuplicate.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+/**
+ * This function remove duplicates string value of an array.
+ * It removes also null, undefined or non string values.
+ *
+ * @param {array} array Array of values.
+ * @ignore
+ */
+const removeDuplicate = (array) => {
```
Comment (mhaertwig) [inline src/common/utils/removeDuplicate.js:8]: For readability, please do this directly in plase.

Comment (mhaertwig) [reply src/common/utils/removeDuplicate.js:8]: ... or the mixin

**Comment by mhaertwig** [comment]
Really nice job, thanks.

**Review by mhaertwig — CHANGES_REQUESTED** [review]

File: src/common/mixins/MapMixin.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    removeMobilityControl(control) {
+      // eslint-disable-next-line no-param-reassign
+      control.map = null;
+      this.mobilityControls = this.mobilityControls.filter(
+        (c) => c !== control,
+      );
+    }
+
+    /** Documentation in inherited classes */
+    addControl(control, position) {
```
[the remark this answers is not in the document]
Comment (oterral) [reply src/common/mixins/MapMixin.js:85]: Why that ? it's more generic like this , you don't have to think which kind of controls you add like for the layers.

File: src/common/utils/removeDuplicate.js (hunk 1 above)
Comment (oterral) [reply src/common/utils/removeDuplicate.js:8]: If you explain me how, without duplicating code, I do it.

File: src/common/mixins/MapMixin.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (friedjoff) [reply src/common/mixins/MapMixin.js:85]: Let's try to be consistent with `addLayer`.

File: src/common/utils/removeDuplicate.js (hunk 1 above)
Comment (oterral) [reply src/common/utils/removeDuplicate.js:8]: fixed

**Comment by oterral** [comment]
@mhaertwig  utils have beend removed and a common/API class has been created.

**Pull request merged**